pkg/baremetal/v1alpha3/types.go

```diff
@@ -1,13 +1,18 @@
-package v1alpha1
+package v1alpha3
 
 import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
-type BareMetalClusterProviderSpec struct {
-	metav1.TypeMeta `json:",inline"`
+type BareMetalCluster struct {
```

Comment:
As the BareMetal provider is an infrastructure provider i think we should implement `status` as `BareMetalStatus` to move it to being a better capi citizen.  In that status we need at least `ready` and `controlPlaneEndpoint` as documented here: https://cluster-api.sigs.k8s.io/developer/architecture/controllers/cluster.html

Reply:
I've added `status.ready` and `spec.controlPlaneEndpoint`; the latter as a renaming of `spec.apiServer.externalLoadBalancer`.